autoload/mr/recorder.vim

```diff
@@ -73,11 +94,13 @@ function! s:dump(recorder) abort
   endtry
 endfunction
 
-function! s:uniq_f() abort
+function! s:trim_f(deletes) abort
   let seen = {}
   return { k, v ->
         \ has_key(seen, v)
         \   ? 0
-        \   : !empty(extend(seen, { v : 1 }))
+        \   : has_key(a:deletes, v)
+        \       ? 0
+        \       : !empty(extend(seen, { v : 1 }))
```

Comment:
Maybe

```vim
function! s:uniq_f(...) abort
  let seen = a:0 ? copy(a:1) : {}
  return { k, v ->
        \ has_key(seen, v)
        \   ? 0
        \   : !empty(extend(seen, { v : 1 }))
        \}
endfunction
```

Reply:
I see, but it always accept a deletion list, so I implemented it as so: c7f2afd